### DISCUSSION TITLE
StreetComplete libraries

### DISCUSSION THREAD

**Discussion by smichel17** [streetcomplete/StreetComplete#3300] [Repository Maintenance] [answered]
In https://github.com/streetcomplete/StreetComplete/issues/3294#issuecomment-922369066, I mentioned pulling out some functionality into a general-purpose library… and then I remembered that there already are several such libraries which have been moved out of the SC code base:

https://github.com/streetcomplete/StreetComplete/blob/29e7ab61d1b0a0295f9804698f38ce7e4a1ccfc6/app/build.gradle.kts#L147-L155

@westnordost I'm wondering how closely they are tied to StreetComplete. Is anyone else using them (that you know about)? Do you *think* of them as separate libraries, which you would continue to maintain if you stopped SC development for some reason, or parts of SC which you're making usable by others (or something else)?

**Comment by westnordost** [comment, marked as answer]
I created osmapi before I started working on StreetComplete. I think of them as separate libraries and they are maintained independently of the app.

Yet, I know no other project that uses them.

Though, `countryboundaries` is used by Vescpucci.

**Comment by smichel17** [comment]
Just for reference in case anyone stumbles across this discussion in the future, **[streetcomplete-mapstyle](https://github.com/streetcomplete/streetcomplete-mapstyle)** has its own repo because it is primarily maintained by @ENT8R, not @westnordost. (You can otherwise find this info in #183, linked from the mapstyle readme.)

**Reply by matkoniecz** [reply]
It is also linked from https://github.com/streetcomplete/StreetComplete/blob/master/CONTRIBUTING.md#streetcomplete-related-projects 

> streetcomplete-mapstyle by @ENT8R maintaining the mapstyle of StreetComplete
